core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -267,7 +334,77 @@ public void gatherAssets(AssetDirectory directory) {
     jumpSound = directory.getEntry("jump", SoundEffect.class);
 
     // This represents the level but does not BUILD it
-    levelFormat = directory.getEntry(levels.get(1), JsonValue.class);
+    this.levelKey = selectedLevelKey;
+    levelFormat = directory.getEntry(selectedLevelKey, JsonValue.class);
+
+    // Create relevant click listeners
+    initClickListeners();
+    initGameMenus();
+
+    // Setup inputs
+    gameInputController = InputController.getInstance();
+    Gdx.input.setInputProcessor(gameInputController.getInputProcessor());
+  }
+
+  /**
+   * Initializes relevant click listeners for this controller
+   */
+  public void initClickListeners() {
```

Comment:
I really appreciate you separating out your code and organizing it this way! It was really readable for me and it's cool to see how the UI implementation works :)